day_5_supply_stacks/stack_parser.rb

```diff
@@ -0,0 +1,78 @@
+require 'minitest/autorun'
+
+class StackParser
+  def initialize(stack_input)
+    @stack_input = stack_input
+    @stack_rows = stack_rows
+  end
+
+  def stack_columns
+    raw_stack_columns = transpose_stack_rows
+    remove_empty_crates_from(raw_stack_columns)
+  end
+
+  private
+
+  attr_reader :stack_input, :stack_row
+
+  def transpose_stack_rows
+    rows = stack_rows.size
+    columns = stack_rows.max_by(&:size).size
+
+    raw_stack_columns = Array.new(columns) { Array.new(rows) }
+
+    stack_rows.each_with_index do |row, i|
+      row.each_with_index do |crate, j|
+        raw_stack_columns[j][i] = crate
+      end
+    end
+
+    raw_stack_columns
+  end
+
+  def remove_empty_crates_from(raw_stack_columns)
+    raw_stack_columns.map do |column|
+      column.delete_if { |crate| crate.nil? || crate.empty? }
```

Comment:
btw. in Rails 6.1+, this could be simply

```ruby
column.compact_blank
```

https://andycroll.com/ruby/compact_blank-remove-empty-strings-from-array-and-hash/